src/Mindee/Input/Compressor.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Docnet.Core;
+using Docnet.Core.Models;
+using Docnet.Core.Readers;
+using Microsoft.Extensions.Logging;
+using Mindee.Exceptions;
+using SkiaSharp;
+
+namespace Mindee.Input
+{
+    /// <summary>
+    /// Compressor static class to handle image and PDF compression.
+    /// </summary>
+    public static class Compressor
+    {
+        private static byte[] CompressImage(SKBitmap original, int quality, int finalWidth, int finalHeight)
+        {
+            using var image = SKImage.FromBitmap(original);
+            using var compressedBitmap = SKBitmap.FromImage(image);
+            using var finalImage =
+                compressedBitmap.Resize(new SKImageInfo(finalWidth, finalHeight), SKFilterQuality.Low);
+            return finalImage.Encode(SKEncodedImageFormat.Jpeg, quality).ToArray();
+        }
+
+
+        /// <summary>
+        /// Resize and/or compress an image using SkiaSharp.
+        /// </summary>
+        /// <param name="imageData">Byte sequence representing the content of the image.</param>
+        /// <param name="quality">Quality of the final file.</param>
+        /// <param name="maxWidth">Maximum width. If not specified, the horizontal ratio will remain the same.</param>
+        /// <param name="maxHeight">Maximum height. If not specified, the vertical ratio will remain the same</param>
```

Comment:
ratio must be maintained in all cases

Reply:
Yeah it's really confusing, but it does maintain ratio. I'll try to make it a bit more explicit

Reply:
not seeing any changes